### PR TITLE
JBIDE-23321 Integration tests for OpenShift Node.js debugger

### PR DESCRIPTION
**Pull request by psrna**

### CODE REVIEW

File: tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+		new WaitWhile(new ShellWithTextIsAvailable(OpenShiftLabel.Shell.IMPORT_APPLICATION));
+
+		// setup server adapter
+		explorer.activate();
+		this.project.getService(OpenShiftResources.NODEJS_SERVICE).select();
+		new ContextMenu(OpenShiftLabel.ContextMenu.NEW_ADAPTER_FROM_EXPLORER).select();
+		new DefaultShell(OpenShiftLabel.Shell.SERVER_ADAPTER_SETTINGS);
+		new FinishButton().click();
+
+		new WaitWhile(new ShellWithTextIsAvailable(""), TimePeriod.LONG);
```
Comment (rawagner) [inline tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java]: Shouldnt OpenShiftLabel.Shell.SERVER_ADAPTER_SETTINGS be here ?

File: tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+		this.adapter.select();
+		new ContextMenu("Restart in Debug").select();
+		new WaitWhile(new JobIsRunning(), TimePeriod.VERY_LONG);
+
+	}
+
+	@After
+	public void clean() {
+		ShellHandler.getInstance().closeAllNonWorbenchShells();
+		new ProjectExplorer().deleteAllProjects();
```
Comment (rawagner) [inline tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java]: You should make sure that ProjectExplorer is open

File: tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+		this.adapter = new ServerAdapter(Version.OPENSHIFT3, OpenShiftResources.NODEJS_SERVICE);
+		this.adapter.select();
+		new ContextMenu("Restart in Debug").select();
+		new WaitWhile(new JobIsRunning(), TimePeriod.VERY_LONG);
+
+	}
+
+	@After
+	public void clean() {
+		ShellHandler.getInstance().closeAllNonWorbenchShells();
```
Comment (rawagner) [inline tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java]: Is this a good idea ? If there is some unexpected shell after test, then it should fail, not close silently and continue

File: tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+			new ServerAdapter(Version.OPENSHIFT3, OpenShiftResources.NODEJS_SERVICE).delete();
+		} catch (OpenShiftToolsException ex) {
+			// do nothing, adapter does not exists
+		}
+	}
+
+	@Test
+	public void testDebuggerStopsAtBreakpoint() throws CoreException, IOException {
+		// TODO use OpenShiftResources.NODEJS_GIT_NAME (once merged)
+		new ProjectExplorer().getProject("nodejs-ex").getProjectItem("server.js").open();
```
Comment (rawagner) [inline tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java]: You should make sure that ProjectExplorer is open

File: tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+		} catch (WaitTimeoutExpiredException e) {
+			Assert.fail("Debugger hasn't stopped on breakpoint");
+		}
+
+	}
+
+	@Test
+	public void testVariablesAvailableInView() throws CoreException, IOException {
+		// TODO use OpenShiftResources.NODEJS_GIT_NAME (once merged)
+		new ProjectExplorer().getProject("nodejs-ex").getProjectItem("server.js").open();
```
Comment (rawagner) [inline tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java]: You should make sure that ProjectExplorer is open

File: tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java (hunk 1 above)
Comment (psrna) [inline tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java]: This is some kind of hack I've discovered in existing os tests. If I replace it, it doesn't work.

File: tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java (hunk 2 above)
Comment (psrna) [inline tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java]: thx, fixed

File: tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java (hunk 4 above)
Comment (psrna) [inline tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java]: thx, fixed

File: tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java (hunk 5 above)
Comment (psrna) [inline tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java]: thx, fixed

File: tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java (hunk 3 above)
Comment (psrna) [inline tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java]: The test will probably fail anyway. But I wanted to make sure its clean for the following tests. Maybe I should put it into @Before?

**Comment by psrna** [comment]
@rhopp Ready to be merged. Please have a look. Thx.

**Comment by psrna** [comment]
testPR

**Comment by psrna** [comment]
@rawagner all is green. Please merge. Thx.

**Pull request merged**